### ISSUE TITLE
Creating a roof on top of a water well destroys water well, replacing it with plank floor

### ISSUE THREAD

**Issue by mkrutov** [CleverRaven/Cataclysm-DDA#53394]
### Describe the bug

n/a

### Steps to reproduce

1. Build water well.
2. Surround it with walls
3. Put a roof on top of it

### Expected behavior

Roof should not replace water well with floor.

### Versions and configuration

These are the outputs for the experimental build of commit 00a0caa
Linux

**Comment by NetSysFire** [comment]
Can reproduce. I assume this happens because the well (`t_covered_well`) is a terrain, just like the floor. It is not furniture.

**Comment by PatrikLundell** [comment]
Yes. This is the result of the design where roofs and floors are built together, rather than being separate entities.
